src/evict/evict_page.c

```diff
@@ -255,20 +256,28 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint8_t previous_state, uint32
 err:
         if (!closing)
             __evict_exclusive_clear(session, ref, previous_state);
-
-        if (time_start != 0) {
-            time_stop = __wt_clock(session);
+        session->evict_timeline.evict_finish = __wt_clock(session);
+        eviction_time = WT_CLOCKDIFF_US(
+          session->evict_timeline.evict_finish, session->evict_timeline.evict_start);
+        if (LF_ISSET(WT_EVICT_CALL_URGENT)) {
             if (force_evict_hs)
                 WT_STAT_CONN_INCR(session, cache_eviction_force_hs_fail);
             WT_STAT_CONN_INCR(session, cache_eviction_force_fail);
-            WT_STAT_CONN_INCRV(
-              session, cache_eviction_force_fail_time, WT_CLOCKDIFF_US(time_stop, time_start));
+            WT_STAT_CONN_INCRV(session, cache_eviction_force_fail_time, eviction_time);
         }
 
         WT_STAT_CONN_DATA_INCR(session, cache_eviction_fail);
     }
 
 done:
+    if (eviction_time > 60 * WT_MILLION)
+        __wt_verbose_warning(session, WT_VERB_EVICT,
+          "Eviction took more than 1 minute. Building disk image took %" PRIu64
```

Comment:
Can you include how long the eviction takes place also? The message tells only that it took more than 1 minute.